tests/legacy/legacy-literal.json

```diff
@@ -0,0 +1,94 @@
+[
+  {
+    "given": {
+      "foo": [
+        {
+          "name": "a"
+        },
+        {
+          "name": "b"
+        }
+      ],
+      "bar": {
+        "baz": "qux"
+      }
+    },
+    "cases": [
+      {
+        "expression": "`foo`",
+        "result": "foo"
+      },
+      {
+        "comment": "Double quotes must be escaped.",
+        "expression": "`foo\\\"quote`",
+        "result": "foo\"quote"
+      },
+      {
+        "expression": "`✓`",
+        "result": "✓"
+      },
+      {
+        "comment": "Double quote in literal",
+        "expression": "`foo\\\"bar`",
+        "result": "foo\"bar"
+      },
+      {
+        "expression": "`1\\``",
+        "result": "1`"
+      },
+      {
+        "comment": "Multiple literal expressions with escapes",
+        "expression": "`\\\\`.{a:`b`}",
+        "result": {
+          "a": "b"
+        }
+      }
+    ]
+  },
+  {
+    "comment": "Literals",
+    "given": {
+      "type": "object"
+    },
+    "cases": [
+      {
+        "expression": "`foo`",
+        "result": "foo"
+      },
+      {
+        "expression": "`      foo`",
+        "result": "foo"
```

Comment:
Oh, interesting! I don't think there's any documentation regarding trimming whitespace in backtick literal legacy fallback behavior, and [JEP-12 Rationale](https://github.com/jmespath-community/jmespath.spec/blob/main/jep-012-raw-string-literals.md#rationale) suggests removal of leading and trailing whitespace (without a precise definition thereof) while implementations vary a bit but none act like that:
* [jmespath.py trims only leading whitespace](https://github.com/jmespath-community/jmespath.py/blob/bbe7300c60056f52413603cf3e2bcd0b6afeda3d/jmespath/lexer.py#L168), implicitly meaning Unicode property White_Space=Yes.
* [jmespath.js trims only leading whitespace](https://github.com/jmespath/jmespath.js/blob/f5f85697449b517f3f41969192a1d9367c4a0465/jmespath.js#L437), implicitly meaning Unicode property White_Space=Yes **OR** U+FEFF ZERO WIDTH NO-BREAK SPACE and **BUT NOT** U+0085 NEXT LINE.
* [jmespath.rb trims only leading whitespace](https://github.com/jmespath/jmespath.rb/blob/b2088367c4b6e72d6e24dbc02e2653a2b90a265c/lib/jmespath/lexer.rb#L337), implicitly meaning **ASCII** characters with Unicode property White_Space=Yes (i.e., U+0009 through U+000D plus U+0020 SPACE).
* [jmespath.php doesn't trim anything](https://github.com/jmespath/jmespath.php/blob/9b87907a81b87bc76d19a7fb2d61e61486ee9edb/src/Lexer.php#L434)

([go-jmespath](https://github.com/jmespath/go-jmespath/blob/b0104c826a24d7c202da8e1dbfb9ea4cfa35d774/parser.go#L321) and [jmespath.rs](https://github.com/jmespath/jmespath.rs/blob/93a173bbffe0c6012af9ce031d471fe3fbf244a9/jmespath/src/lexer.rs#L291-L294) do not support the fallback behavior AFAICT)

If this is codified in the compliance test suite, it should also be specified as part of https://github.com/jmespath-community/jmespath.spec/issues/133 .

Reply:
As I write [there](https://github.com/jmespath-community/jmespath.spec/issues/133#issuecomment-1314578465), I'm not sure I agree that the spec needs to be concerned with specifying legacy behaviour. However, I would be comfortable including this in any addendum made to JEP-12.